### PR TITLE
refactor: 💡 Move e2e tests to new folder

### PR DESCRIPTION
**Pull request by ZedLi**
# Description
The goal of this PR is to make it easier to share the setup between admin and desktop E2E tests. We want to leverage the helpers from admin UI for setting things up but the current folder structure makes it difficult to do.

I moved both admin UI and desktop e2e tests to a new folder at the root. I made it part of the overall mono-repo we have as a separate workspace so it shares the `yarn.lock` and node modules.

Feel free to suggest different folder structures, this doesn't have to be the final setup of our e2e tests.

## Todo
- [x] Fix license checker for new e2e folder package, thinks it's unlicensed
- [x] Change desktop tests to mjs (and then change package.json to module)
- [ ] Fix formatter/linter and run it for everything (maybe a separate PR to make it easier to review)

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
I like this approach. Being able to reuse the helpers for Admin and Desktop is a huge benefit and this will likely help keep e2e tests and other levels of tests separated with clear lines.

File: e2e-tests/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "format:js": "prettier --write './**/*.{js,mjs}'",
+    "e2e:admin": "yarn playwright test --config ./admin/playwright.config.mjs",
+    "e2e:admin:ce:aws": "yarn run e2e:admin --grep \"(?=.*@ce)(?=.*@aws)\"",
+    "e2e:admin:ce:docker": "yarn run e2e:admin --grep \"(?=.*@ce)(?=.*@docker)\"",
+    "e2e:admin:ent:aws": "yarn run e2e:admin --grep \"(?=.*@ent)(?=.*@aws)\"",
+    "e2e:admin:ent:docker": "yarn run e2e:admin --grep \"(?=.*@ent)(?=.*@docker)\"",
+    "e2e:admin:dev:chrome": "yarn run e2e:admin --project=chromium --headed",
+    "e2e:admin:dev:firefox": "yarn run e2e:admin --project=firefox --headed",
+    "e2e:admin:dev:safari": "yarn run e2e:admin --project=webkit --headed",
+    "e2e:desktop": "yarn workspace desktop build && yarn run e2e:desktop:dev",
```
[the remark this answers is not in the document]
Comment (cameronperera) [reply e2e-tests/package.json:23]: Oh that is a good change. I will be honest that I didn't realize `yarn workspace` was a thing.

File: e2e-tests/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "e2e:desktop:dev": "npx playwright test --config ./desktop/playwright.config.js --reporter=html"
```
[the remark this answers is not in the document]
Comment (cameronperera) [reply e2e-tests/package.json:24]: Now that we are in a dedicated e2e directory, I would say we can remove the e2e but I don't have a strong preference against it.

[the remark this answers is not in the document]
Comment (moduli) [reply e2e-tests/package.json:24]: I'm cool with removing it. It seems redundant in this new directory.

**Review by moduli — COMMENTED** [review]
I'm on board with these changes. Successfully tested running all of the Admin UI tests in this new structure.

Can re-review once the PR is out of draft.

File: e2e-tests/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "format": "concurrently \"npm:format:*\" --names \"format:\"",
+    "format:js": "prettier --write './**/*.{js,mjs}'",
+    "admin": "yarn playwright test --config admin/playwright.config.js",
+    "admin:ce:aws": "yarn run e2e:admin --grep \"(?=.*@ce)(?=.*@aws)\"",
+    "admin:ce:docker": "yarn run e2e:admin --grep \"(?=.*@ce)(?=.*@docker)\"",
+    "admin:ent:aws": "yarn run e2e:admin --grep \"(?=.*@ent)(?=.*@aws)\"",
+    "admin:ent:docker": "yarn run e2e:admin --grep \"(?=.*@ent)(?=.*@docker)\"",
+    "admin:dev:chrome": "yarn run e2e:admin --project=chromium --headed",
+    "admin:dev:firefox": "yarn run e2e:admin --project=firefox --headed",
+    "admin:dev:safari": "yarn run e2e:admin --project=webkit --headed",
```
Comment (moduli) [inline e2e-tests/package.json:23]: Need to remove the `e2e` from those subcommands
```suggestion
    "admin:ce:aws": "yarn run admin --grep \"(?=.*@ce)(?=.*@aws)\"",
    "admin:ce:docker": "yarn run admin --grep \"(?=.*@ce)(?=.*@docker)\"",
    "admin:ent:aws": "yarn run admin --grep \"(?=.*@ent)(?=.*@aws)\"",
    "admin:ent:docker": "yarn run admin --grep \"(?=.*@ent)(?=.*@docker)\"",
    "admin:dev:chrome": "yarn run admin --project=chromium --headed",
    "admin:dev:firefox": "yarn run admin --project=firefox --headed",
    "admin:dev:safari": "yarn run admin --project=webkit --headed",
```

File: e2e-tests/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "desktop": "yarn workspace desktop build && yarn run e2e:desktop:dev",
```
Comment (moduli) [inline e2e-tests/package.json:24]: Here, too?
```suggestion
    "desktop": "yarn workspace desktop build && yarn run desktop:dev",
```

Comment (ZedLi) [reply e2e-tests/package.json]: Fixed

**Review by DhariniJeeva — APPROVED** [review]
this looks great, thanks for setting this up!!

**Review by moduli — APPROVED** [review]

**Pull request merged**